id_infrastructure/firestore_uuid_table.py

```diff
@@ -119,11 +134,18 @@ def uuid_to_data(self, uuid_to_lookup):
         raise LookupError() 
 
     def uuid_to_data_batch(self, uuids_to_lookup):
+        # Serve the request from the cache if possible, saving network request time + Firestore read costs
+        uuids_to_lookup = set(uuids_to_lookup)
+        if len(uuids_to_lookup - set(self._mappings_cache.values())) == 0:
```

Comment:
if we are partly missing some uids e.g if running the pipeline twice concurrently(which is rare) we should probably return the uuids available in cache and only look up the rest from firestore.

Reply:
As far as I could tell from the documentation, there isn't an efficient way to look up a list of ids from Firestore. The best I could find is 'where in' but that only supports 10 at a time so we might need to make thousands of round trips, which I thought would be very slow.

Reply:
Okay, thanks